### PR TITLE
Added build config and test structure for unsigned int keys. Note tha…

### PR DESCRIPTION
**Pull request by ireed**
…t this does not contain the actual code/support for uint32 keys - that will be the next commit.

### CODE REVIEW

**Comment by ireed** [comment]
On hold until travis issue is resolved.
It works in my branch (https://travis-ci.org/ireed/ISx/builds/257601360) but not here, for some reason.

File: SHMEM/isx.c [hunk 1]
```diff
@@ -867,7 +867,7 @@ static void wait_my_turn()
 
 static void my_turn_complete()
 {
-  const int my_rank = shmem_my_pe();
+  const uint32_t my_rank = shmem_my_pe();
```
Comment (jdinan) [inline SHMEM/isx.c:870]: The return type of ```shmem_my_pe()``` is ```int```, so the type of ```my_rank``` should also be ```int```.

**Comment by jdinan** [comment]
This should also be tested with a key type of ```uint64_t``` to check for bugs, since ```int``` and ```uint32_t``` will be the same size on all machines where we test.

File: SHMEM/params.h [hunk 2]
```diff
@@ -43,7 +44,11 @@ POSSIBILITY OF SUCH DAMAGE.
 
 // The data type used for the keys
 // If you change this, you will have to change the SHMEM API calls used
+#ifdef UINT32_KEYS
+typedef uint32_t KEY_TYPE;
+#else
 typedef int KEY_TYPE;
```
Comment (jdinan) [inline SHMEM/params.h:50]: `int` and `uint32_t` are the same size.  Do you want to support different size keys?  Also, I would suggest that all key types have the same signed-ness (signed or unsigned).

**Comment by jdinan** [comment]
@ireed What's the status of this PR?

File: SHMEM/params.h [hunk 3]
```diff
@@ -72,6 +77,8 @@ typedef int KEY_TYPE;
 // to keep the BUCKET_WIDTH constant per PE.
 #ifdef DEBUG
 #define DEFAULT_MAX_KEY (32uLL)
+#elif defined(UINT32_KEYS)
+#define DEFAULT_MAX_KEY (unsigned long long) UINT_MAX
```
Comment (jdinan) [inline SHMEM/params.h:81]: Use `UINT32_MAX` from `stdint.h`

File: SHMEM/params.h [hunk 4]
```diff
@@ -72,6 +77,8 @@ typedef int KEY_TYPE;
...
 #else
 #define DEFAULT_MAX_KEY (unsigned long long)(1uLL<<28uLL)
```
Comment (jdinan) [inline SHMEM/params.h:83]: 🤦‍♂️

**Comment by jdinan** [comment]
This PR appears to have gone stale, closing for now.  Please feel free to repost when the issues noted here are resolved.  If you want to continue testing with Travis, you can enable testing on your fork on the repository through the Travis website.

**Pull request closed without merging**